custom_components/lutron_caseta_pro/fan.py

```diff
@@ -190,15 +190,13 @@ def supported_features(self) -> int:
     def update_state(self, value):
         """Update internal state and fan speed."""
         self._is_on = value > SPEED_MAPPING[SPEED_OFF]
-        if value in range(SPEED_MAPPING[SPEED_MEDIUM_HIGH] + 1, SPEED_MAPPING[SPEED_HIGH] + 1):
+        if value > SPEED_MAPPING[SPEED_MEDIUM_HIGH] and value <= SPEED_MAPPING[SPEED_HIGH]:
             self._speed = SPEED_HIGH
-        elif value in range(SPEED_MAPPING[SPEED_MEDIUM] + 1, SPEED_MAPPING[SPEED_MEDIUM_HIGH] + 1):
-            # 51% - 55% are missing from Lutron integration protocol
-            # we will treat as medium_high
+        elif value > SPEED_MAPPING[SPEED_MEDIUM] and value <= SPEED_MAPPING[SPEED_MEDIUM_HIGH]:
             self._speed = SPEED_MEDIUM_HIGH
-        elif value in range(SPEED_MAPPING[SPEED_LOW] + 1, SPEED_MAPPING[SPEED_MEDIUM] + 1):
+        elif value > SPEED_MAPPING[SPEED_LOW] and value <= SPEED_MAPPING[SPEED_MEDIUM]:
             self._speed = SPEED_MEDIUM
-        elif value in range(SPEED_MAPPING[SPEED_OFF] + 1, SPEED_MAPPING[SPEED_LOW] + 1):
+        elif value > SPEED_MAPPING[SPEED_OFF] and value <= SPEED_MAPPING[SPEED_LOW]:
```

Comment:
Since the range() operator is unavailable, I think these can be compounded like this:
```python
        if SPEED_MAPPING[SPEED_MEDIUM_HIGH] < value <= SPEED_MAPPING[SPEED_HIGH]:
            self._speed = SPEED_HIGH
        elif SPEED_MAPPING[SPEED_MEDIUM] < value <= SPEED_MAPPING[SPEED_MEDIUM_HIGH]:
            self._speed = SPEED_MEDIUM_HIGH
        elif SPEED_MAPPING[SPEED_LOW] < value <= SPEED_MAPPING[SPEED_MEDIUM]:
            self._speed = SPEED_MEDIUM
        elif SPEED_MAPPING[SPEED_OFF] < value <= SPEED_MAPPING[SPEED_LOW]:
            self._speed = SPEED_LOW
        elif value == SPEED_MAPPING[SPEED_OFF]:
            self._speed = SPEED_OFF
```

Reply:
Ahh good point - that is significantly shorter and easier to read. I updated in the latest commit.